CMakeLists.txt

```diff
@@ -153,35 +172,53 @@ include(CPack)
 message(STATUS "Setting link libraries")
 
 function(protocol protoPath protoName external)
-    if (external)
+    if(external)
         execute_process(
```

Comment:
using `add_custom_command()` here to generate these files into `CMAKE_CURRENT_BINARY_DIR` would be nice. this way we could add them to an interface library target that any dependents can `target_link_libraries()` to.